zcash_client_backend/src/data_api/testing/transparent.rs

```diff
@@ -243,5 +272,128 @@ where
         .expiry_height();
     st.wallet_mut().update_chain_tip(expiry_height).unwrap();
 
-    check_balance(&st, 0, value);
+    check_balance(&st, 0, &zero_or_one_conf_value);
+    // we currently treat min_confirmation the same regardless they are 0 or 1
+    check_balance(&st, 1, &zero_or_one_conf_value);
+}
+
+/// This test attempts to verify that transparent funds spendability is 
+/// accounted for properly given the different [`min_confirmations`] values
+pub fn transparent_balance_spendability<DSF>(dsf: DSF, cache: impl TestCache)
+where
+    DSF: DataStoreFactory,
+{
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(dsf)
+        .with_block_cache(cache)
+        .with_account_from_sapling_activation(BlockHash([0; 32]))
+        .build();
+
+    let account = st.test_account().cloned().unwrap();
+    let uaddr = st
+        .wallet()
+        .get_current_address(account.id())
+        .unwrap()
+        .unwrap();
+    let taddr = uaddr.transparent().unwrap();
+
+    // Initialize the wallet with chain data that has no shielded notes for us.
+    let not_our_key = ExtendedSpendingKey::master(&[]).to_diversifiable_full_viewing_key();
+    let not_our_value = NonNegativeAmount::const_from_u64(10000);
+    let (start_height, _, _) =
+        st.generate_next_block(&not_our_key, AddressType::DefaultExternal, not_our_value);
+    for _ in 1..10 {
+        st.generate_next_block(&not_our_key, AddressType::DefaultExternal, not_our_value);
+    }
+    st.scan_cached_blocks(start_height, 10);
+
+    // TODO: abstract this function so it's not duplicated among different tests. 
```

Comment:
Since this PR is going from one to two uses of this code (i.e. it's introducing the duplication), and since it looks straightforward to move it out to a private function because it only closes over `account` and `taddr`, please do so now.

Reply:
the amount of generics that are involved in that apparently innocent function... oh dear FSM! 😅

I'll do my best!